### PR TITLE
User Profile - Add rest spec files and tests

### PR DESCRIPTION
**Pull request by ywangd**
This PR adds rest spec files and tests for existing user profile APIs.
* Get User Profile
* Activate User Profile
* Update application data for Profile

It also includes stub doc pages so that it does not build doc builds.
But the actual content of docs will be handled in a separate PR.

### CODE REVIEW

File: x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/user_profile/10_basic.yml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  # Data should be preserved
+  - do:
+      security.get_user_profile:
+        uid: "$profile_uid"
+        data: "*"
+
+  - length: { $body: 1 }
+  - is_true: "$profile_uid"
+  - set: { $profile_uid: profile }
+  - match: { $profile.data: { "app1": { "theme": "default" }, "app2": { "theme": "dark"} } }
```
Comment (sethmlarson) [inline x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/user_profile/10_basic.yml:104]: Can we also add test cases for errors (ie when a profile doesn't exist?)

**Review by sethmlarson — COMMENTED** [review]
The API specs look good, could you post/send me an example of all the response bodies? Especially curious about the `_doc` in `security.activate_user_profile`.

File: x-pack/docs/en/rest-api/security/update-user-profile-data.asciidoc [hunk 2]
```diff
@@ -0,0 +1,10 @@
+[role="xpack"]
+[[security-api-update-user-profile-data]]
+=== Activate user profile API
```
[the remark this answers is not in the document]
Comment (ywangd) [reply x-pack/docs/en/rest-api/security/update-user-profile-data.asciidoc]: Thanks. Updated.

File: x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/user_profile/10_basic.yml (hunk 1 above)
Comment (ywangd) [reply x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/user_profile/10_basic.yml:104]: Good point. Added now.

**Comment by ywangd** [comment]
@elasticmachine update branch

**Comment by ywangd** [comment]
@elasticmachine update branch

**Comment by ywangd** [comment]
@elasticmachine update branch

File: x-pack/docs/en/rest-api/security/activate-user-profile.asciidoc [hunk 3]
```diff
@@ -0,0 +1,10 @@
+[role="xpack"]
+[[security-api-activiate-user-profile]]
+=== Activate user profile API
+++++
+<titleabbrev>Activate user profile</titleabbrev>
+++++
+
+Creates or updates the user profile on behalf of another user
+
+coming::[8.1.0]
```
Comment (albertzaharovits) [inline x-pack/docs/en/rest-api/security/activate-user-profile.asciidoc:10]: It's 8.2.0 now :(

File: x-pack/docs/en/rest-api/security/activate-user-profile.asciidoc [hunk 4]
```diff
@@ -0,0 +1,10 @@
+[role="xpack"]
+[[security-api-activiate-user-profile]]
+=== Activate user profile API
+++++
+<titleabbrev>Activate user profile</titleabbrev>
+++++
+
+Creates or updates the user profile on behalf of another user
```
Comment (albertzaharovits) [inline x-pack/docs/en/rest-api/security/activate-user-profile.asciidoc:8]: I find the "on behalf" bit redundant.

File: x-pack/docs/en/rest-api/security/get-user-profile.asciidoc [hunk 5]
```diff
@@ -0,0 +1,10 @@
+[role="xpack"]
+[[security-api-get-user-profile]]
+=== Get user profile API
+++++
+<titleabbrev>Get user profile</titleabbrev>
+++++
+
+Retrieves user profile for the given unique ID
```
Comment (albertzaharovits) [inline x-pack/docs/en/rest-api/security/get-user-profile.asciidoc:8]: nit:
```suggestion
Retrieves a user's profile given the unique profile ID
```

**Review by albertzaharovits — APPROVED** [review]
LGTM

**Review by sethmlarson — APPROVED** [review]
Specs look good to me!

File: x-pack/docs/en/rest-api/security/activate-user-profile.asciidoc (hunk 3 above)
Comment (ywangd) [reply x-pack/docs/en/rest-api/security/activate-user-profile.asciidoc:10]: Updated to 8.2.0

File: x-pack/docs/en/rest-api/security/activate-user-profile.asciidoc (hunk 4 above)
Comment (ywangd) [reply x-pack/docs/en/rest-api/security/activate-user-profile.asciidoc:8]: It's to emphasize that there are two users involved in this API, the caller and the target user for creating the profile document. I'll leave it as is and we can come back to it in #83495

**Pull request merged**